src/waveform/waveform.h

```diff
@@ -31,6 +31,13 @@ union WaveformData {
 
 class Waveform {
   public:
+    enum class SaveState {
+        Virgin = 0,
+        Changing, 
```

Comment:
The state "Changing" is unused. Currently it is only set and that part of the code even suffers from threading issues (see my comment).